db/migrate/20160229082515_create_attachment_references.rb

```diff
@@ -0,0 +1,17 @@
+class CreateAttachmentReferences < ActiveRecord::Migration
+  def change
+    create_table :attachment_references do |t|
+      t.references :attachable, polymorphic: true
+      t.references :attachment, null: false
+      t.string :name, null: false
+      t.datetime :expiry_at
+
```

Comment:
Would you call this expiry_at or expires_at? sorry, I didn't catch this earlier.